pkg/util/docker/metadata.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build docker
+
+package docker
+
+import (
+	"context"
+
+	log "github.com/cihub/seelog"
+	"github.com/docker/docker/api/types/swarm"
+
+	"github.com/DataDog/datadog-agent/pkg/metadata/host/container"
+)
+
+func init() {
+	container.RegisterMetadataProvider("docker", getMetadata)
+}
+
+func getMetadata() map[string]string {
+	metadata := make(map[string]string)
+	du, err := GetDockerUtil()
+	if err != nil {
+		log.Debugf("Unable to collect Docker host metadata: %s", err)
+		return metadata
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), du.queryTimeout)
```

Comment:
To avoid delaying the metadata generation too much, could we use a 1 second timeout for all requests in this method?

As for GetSwarmState, if we think it will be reused, we could change it to a method accepting a `types.Info` object, like we do for [parseStorageStatsFromInfo](https://github.com/DataDog/datadog-agent/blob/5295276b83de1490836b44ce0d488a83e1aa53f9/pkg/util/docker/storage.go#L80). That way:
  - we can unit-test the parsing
  - we can retrieve the info with a one-second timeout for this specific use
  - later we could extract other tags from it without multiplying requests

Reply:
both swarm and server version are in `types.Info` simplifying the code 👍